File: Go/src/main/QuestionBank/Solution001_100/Solution1.go
package main

func main() {

}

//func twoSum(nums []int, target int) []int {
//	for i := 0; i < len(nums); i++ {
//		for j := i + 1; j < len(nums); j++ {
//			if nums[i]+nums[j] == target {
//				return []int{i, j}
//			}
//		}
//	}
//	return nil
//}

func twoSum(nums []int, target int) []int {
	var hashTable = make(map[int]int)

	for i, x := range nums {
		if p, ok := hashTable[target-x]; ok {
			return []int{p, i}
		}
		hashTable[x] = i
	}

	for i, num := range nums {
		if p, ok := hashTable[target-num]; ok {
			return []int{p, i}
		}
		hashTable[num] = i
	}
	return nil
}

File: Java/src/main/java/QuestionBank/Solution001_100/Solution1.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution1 {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            int temp = target - nums[i];
            for (int j = i + 1; j < nums.length; j++) {
                if (temp == nums[j]) {
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{0, 0};
    }

    public static void main(String[] args) {
        System.out.println(new Solution1().twoSum(new int[]{1, 1}, 1));
    }
}

